app/src/main/java/com/lucasfagundes/androidudemycourse/feature/random_number/RandomNumberFragment.kt

```diff
@@ -5,36 +5,41 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
 import com.lucasfagundes.androidudemycourse.R
 import com.lucasfagundes.androidudemycourse.databinding.FragmentRandomNumberBinding
-import kotlin.random.Random
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class RandomNumberFragment : Fragment() {
 
     private lateinit var binding: FragmentRandomNumberBinding
+    private val viewModel: RandomNumberViewModel by viewModel()
 
     override fun onCreateView(
         layoutInflater: LayoutInflater,
         container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View? {
+        savedInstanceState: Bundle?,
+    ): View {
         binding = FragmentRandomNumberBinding.inflate(layoutInflater, container, false)
         return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         handleButtonClick()
+        handleObserver()
     }
 
     private fun handleButtonClick() {
         binding.playButton.setOnClickListener {
-            setRandomNumber()
+            viewModel.setRandomNumber()
         }
     }
 
-    private fun setRandomNumber() {
-        val randomNumber = Random.nextInt(0, 20)
-        binding.noNumberChosen.text = getString(R.string.number_answer, randomNumber.toString())
+    private fun handleObserver() {
+        val result = Observer<String> { result ->
+            binding.resultNumberTextView.text = result.toString()
+        }
+        viewModel.result.observe(viewLifecycleOwner, result)
```

Comment:
```suggestion
    private fun handleObserver() {
        val result = Observer<String> { result ->
            binding.resultNumberTextView.text = result.toString()
        }
        viewModel.result.observe(viewLifecycleOwner, result)
```

        viewModel.result.observe(viewLifecycleOwner, Observer<String> { result ->
            binding.resultNumberTextView.text = result.toString()
        })

se tu fizer assim, acho que fica mais legivel e de facil entendimento

Reply:
ao inves de atribuir o observer a uma variavel, vc so passa ele como parametro direto